src/wp-includes/general-template.php

```diff
@@ -3422,6 +3422,121 @@ function wp_resource_hints() {
 	}
 }
 
+/**
+ * Prints resource preloads directives to browsers.
+ *
+ * Gives directive to browsers to preload specific resources that website will
+ * need very soon, this ensures that they are available earlier and are less
+ * likely to block the page's render.
+ *
+ * These performance improving indicators work by using `<link rel"preload">`.
+ *
+ * @link https://developer.mozilla.org/en-US/docs/Web/HTML/Link_types/preload
+ * @link https://web.dev/preload-responsive-images/
+ *
+ * @since 6.1.0
+ */
+function wp_preload_links() {
+	/**
+	 * Filters domains and URLs for resource preloads.
+	 *
+	 * @since 6.1.0
+	 *
+	 * @param array  $urls {
+	 *     Array of resources and their attributes, or URLs to print for resource preloads.
+	 *
+	 *     @type array ...$0 {
+	 *         Array of resource attributes.
+	 *
+	 *         @type string $href        URL to include in resource preloads. Required.
+	 *         @type string $as          How the browser should treat the resource
+	 *                                   (`script`, `style`, `image`, `document`, etc).
+	 *         @type string $crossorigin Indicates the CORS policy of the specified resource.
+	 *         @type string $type        Type of the resource (`text/html`, `text/css`, etc).
+	 *         @type string $media       Accepts media types or media queries. Allows responsive preloading.
+	 *         @type string $imagesizes  Responsive source size to the source Set.
+	 *         @type string $imagesrcset Responsive image sources to the source set.
+	 *     }
+	 * }
+	 */
+	$urls = apply_filters( 'wp_preload_links', array() );
+
+	if ( ! is_array( $urls ) ) {
+		return;
+	}
+
+	$unique_urls = array();
+	foreach ( $urls as $url ) {
+		if ( ! is_array( $url ) ) {
+			continue;
+		}
+
+		$atts = $url;
+		if ( isset( $url['href'] ) ) {
+			$url = $url['href'];
```

Comment:
Don't think resetting/reusing vars is a good idea, ever. I know, it comes from `wp_resource_hints()` however it should be fixed there rather than replicated here. See above.

Reply:
fixed here